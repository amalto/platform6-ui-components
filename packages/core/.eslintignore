dist/
loader/

src/components.d.ts