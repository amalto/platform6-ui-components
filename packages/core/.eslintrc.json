{
  "extends": ["../../.eslintrc.json", "plugin:@stencil/recommended", "plugin:jest/recommended", "plugin:jsx-a11y/recommended"],
  "parserOptions": {
    "project": ["./tsconfig.json", "./tsconfig.lint.json"],
    "ecmaFeatures": {
      "jsx": true // Allows for the parsing of JSX
    }
  },
  "plugins": ["jest"],

  "rules": {
    "jsx-a11y/label-has-associated-control": [
      2,
      {
        "assert": "either"
      }
    ],
    "react/jsx-props-no-spreading": "off",
    "react/no-unknown-property": [2, { "ignore": ["class"] }]
  },
  "settings": {
    "react": {
      "pragma": "h"
    }
  }
}
