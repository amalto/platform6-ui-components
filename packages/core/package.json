{
  "name": "@platform6/components",
  "version": "0.0.1",
  "description": "Platform 6 UI components",
  "repository": {
    "type": "git",
    "url": "https://github.com/amalto/platform6-ui-components"
  },
  "license": "MIT",
  "main": "dist/index.cjs.js",
  "module": "dist/custom-elements/index.js",
  "collection": "dist/collection/collection-manifest.json",
  "collection:main": "dist/collection/index.js",
  "es2015": "dist/esm/index.mjs",
  "es2017": "dist/esm/index.mjs",
  "unpkg": "dist/stencil-component/stencil-component.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/",
    "loader/"
  ],
  "scripts": {
    "build": "stencil build --docs",
    "format": "prettier --write ./",
    "postformat": "npm run lint.fix",
    "generate": "stencil generate",
    "lint": "npm run lint.ts && npm run lint.sass",
    "lint.fix": "npm run lint.ts.fix && npm run lint.sass.fix",
    "lint.sass": "stylelint 'src/**/*.scss'",
    "lint.sass.fix": "npm run lint.sass -- --fix",
    "lint.ts": "eslint --cache --ext .js,.jsx,.ts,.tsx ./",
    "lint.ts.fix": "npm run lint.ts -- --fix",
    "start": "stencil build --docs --dev --watch --serve",
    "test": "stencil test --spec --e2e",
    "test.watch": "stencil test --spec --e2e --watchAll",
    "storybook": "start-storybook -p 6006 -s ./dist --ci",
    "build-storybook": "build-storybook"
  },
  "dependencies": {
    "@creativebulma/bulma-tooltip": "^1.2.0",
    "@fortawesome/fontawesome-svg-core": "^1.2.32",
    "@fortawesome/free-brands-svg-icons": "^5.15.1",
    "@fortawesome/free-regular-svg-icons": "^5.15.1",
    "@fortawesome/free-solid-svg-icons": "^5.15.1",
    "@stencil/core": "2.2.0",
    "bulma": "^0.9.1",
    "bulma-calendar": "^6.0.9",
    "choices.js": "9.0.1",
    "lit-html": "^1.3.0"
  },
  "devDependencies": {
    "@babel/core": "^7.12.3",
    "@stencil/eslint-plugin": "^0.3.1",
    "@stencil/postcss": "^1.0.1",
    "@stencil/react-output-target": "0.0.9",
    "@stencil/sass": "^1.3.2",
    "@storybook/addon-a11y": "^6.0.28",
    "@storybook/addon-actions": "^6.0.28",
    "@storybook/addon-essentials": "^6.0.28",
    "@storybook/addon-links": "^6.0.28",
    "@storybook/html": "^6.0.28",
    "@storybook/preset-scss": "^1.0.3",
    "@storybook/theming": "^6.0.28",
    "@storybook/web-components": "^6.0.28",
    "@types/jest": "^26.0.15",
    "@types/js-beautify": "^1.11.0",
    "@typescript-eslint/eslint-plugin": "^4.6.1",
    "@typescript-eslint/parser": "^4.6.1",
    "autoprefixer": "^9.8.6",
    "babel-loader": "^8.1.0",
    "css-loader": "^5.0.0",
    "eslint-plugin-jest": "^24.1.0",
    "jest": "^26.6.3",
    "jest-cli": "^26.6.3",
    "js-beautify": "^1.13.0",
    "prettier": "^2.1.2",
    "puppeteer": "^5.4.1",
    "react-is": "^17.0.1",
    "sass-loader": "^10.0.5",
    "style-loader": "^2.0.0",
    "typescript": "4.0.5"
  }
}
