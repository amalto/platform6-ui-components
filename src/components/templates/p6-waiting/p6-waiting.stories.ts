import { Components } from "../../../components";
import {
  getElement,
  makeStory,
  Props,
} from "../../../shared/storybook/stories";

const component = "p6-waiting";

export default {
  title: "Templates/Waiting",
  component,
};

const getStoryField = (
  child: string | HTMLElement | HTMLElement[],
  props?: Props<Components.P6Waiting>
): HTMLElement => {
  return getElement(component, child, props);
};

// Default
export const DefaultStory = makeStory({
  builder: (): HTMLElement => getStoryField([]),
});

// Centered
export const CenteredStory = makeStory({
  builder: (): HTMLElement =>
    getStoryField([], {
      style: { height: "280px" } as CSSStyleDeclaration,
    }),
});

// Message
export const MessageStory = makeStory<{
  text: string;
}>({
  args: {
    text: "Loading data...",
  },
  builder: ({ text }): HTMLElement => getStoryField(text),
});

// Message
export const RichMessageStory = makeStory<{
  text: string;
  link: string;
}>({
  args: {
    text: "Loading data...",
    link: "Cancel",
  },
  builder: ({ text, link }): HTMLElement =>
    getStoryField([getElement("span", text), getElement("p6-link", link)]),
});
