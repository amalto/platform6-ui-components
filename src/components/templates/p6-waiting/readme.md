# p6-waiting

<!-- Auto Generated Below -->

## Dependencies

### Depends on

- [p6-spinner](../../atoms/p6-spinner)

### Graph

```mermaid
graph TD;
  p6-waiting --> p6-spinner
  style p6-waiting fill:#f9f,stroke:#333,stroke-width:4px
```

---

_Built with [StencilJS](https://stenciljs.com/)_
