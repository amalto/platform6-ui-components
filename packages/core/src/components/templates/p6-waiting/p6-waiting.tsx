import { Component, h, Host, JSX } from '@stencil/core';

@Component({
  tag: 'p6-waiting',
  styleUrl: 'p6-waiting.scss',
  shadow: true,
})
export class P6Waiting {
  render(): JSX.Element {
    return (
      <Host>
        <p6-spinner />
        <slot />
      </Host>
    );
  }
}
