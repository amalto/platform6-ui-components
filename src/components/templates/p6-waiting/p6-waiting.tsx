import { Component, h, Host } from "@stencil/core";

@Component({
  tag: "p6-waiting",
  styleUrl: "p6-waiting.scss",
  scoped: true,
})
export class P6Waiting {
  render(): JSX.Element {
    return (
      <Host>
        <p6-spinner />
        <slot />
      </Host>
    );
  }
}
