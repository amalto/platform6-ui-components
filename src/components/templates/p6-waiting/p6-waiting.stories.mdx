import { Story, Preview, Meta } from "@storybook/addon-docs/blocks";

import readme from "./readme.md";

<Meta
  title="Templates/Waiting"
  component="p6-waiting"
  parameters={{ notes: readme }}
/>

# Waiting

`p6-waiting` provides a spinner

<Preview>
  <Story name="Default">
    {() => {
      return `<p6-waiting></p6-waiting>`;
    }}
  </Story>
</Preview>

<Story name="Centered" height="300px" parameters={{ docs: { disable: true } }}>
  {() => {
    return '<p6-waiting style="height:280px;"></p6-waiting>';
  }}
</Story>

<Story
  name="With message"
  height="300px"
  parameters={{ docs: { disable: true } }}
>
  {() => {
    return '<p6-waiting style="height:280px;">Loading data...</p6-waiting>';
  }}
</Story>

<Story
  name="With rich message"
  height="300px"
  parameters={{ docs: { disable: true } }}
>
  {() => {
    return '<p6-waiting style="height:280px;">Loading data... <p6-link>Cancel</p6-link></p6-waiting>';
  }}
</Story>
