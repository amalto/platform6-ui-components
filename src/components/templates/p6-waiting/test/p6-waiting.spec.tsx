import { newSpecPage } from '@stencil/core/testing';
import { P6Waiting } from '../p6-waiting';

describe('p6-waiting', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [P6Waiting],
      html: `<p6-waiting></p6-waiting>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('with message', async () => {
    const page = await newSpecPage({
      components: [P6Waiting],
      html: `<p6-waiting>Loading in progress</p6-waiting>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
