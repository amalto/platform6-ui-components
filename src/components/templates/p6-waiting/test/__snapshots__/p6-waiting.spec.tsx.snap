// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`p6-waiting renders 1`] = `
<p6-waiting>
  <mock:shadow-root>
    <p6-spinner></p6-spinner>
    <slot></slot>
  </mock:shadow-root>
</p6-waiting>
`;

exports[`p6-waiting with message 1`] = `
<p6-waiting>
  <mock:shadow-root>
    <p6-spinner></p6-spinner>
    <slot></slot>
  </mock:shadow-root>
  Loading in progress
</p6-waiting>
`;
